### DISCUSSION TITLE
Control with physical buttons

### DISCUSSION THREAD

**Discussion by TB31388** [mikebrady/shairport-sync#1740] [Q&A] [answered]
I have a Bang & Olufsen Beoplay A8 that I have connected to a Raspberyy pi 4 by usb (Beoplay A8 has a built in usb DAC, but only Airplay 1). It works great!

Prior to using Shairport-sync for this I used an old mac mini to get airplay 2.

When I connected it via the mac mini I was able to use the physical buttons on the Beoplay A8 to skip tracks and pause. I can't get this to work with shairport-sync.

Any idea how I can do this with shairport-sync?
![image](https://github.com/mikebrady/shairport-sync/assets/83475766/bbbb508a-2178-4ba1-85c9-675979e6d0bf)

**Comment by mikebrady** [comment]
Thanks for the post. Shairport Sync (“SPS”) treats the A8 strictly as a DAC and doesn’t receive any return information from it, so unfortunately it isn’t possible to get SPS to respond to commands coming from it.

**Reply by TB31388** [reply]
I understand that. Any that this could be implemented in the future? I suspect that there are a number of people that have USB DAC's with physical buttons.

**Reply by mikebrady** [reply]
No plans, I'm afraid.

**Comment by TB31388** [comment, marked as answer]
I got this working by asking ChatGPT. In summary one can use the evtest application to detect button presses. Then write a script (ask chatgpt to do it) to forward the commands received to the D-bus interface of SPS.

**Reply by noelhibbard** [reply]
Impressive use of ChatGPT.

**Reply by lmartindesile** [reply]
any chance to share the script, please. I've tried the same way, without success. being stuck with that d-bus thing

**Reply by TB31388** [reply]
Happy to share the script but please be aware that due to changes in ios buttons currently don't work.
